pandas/tests/arrays/test_integer.py

```diff
@@ -1074,6 +1074,23 @@ def test_cut(bins, right, include_lowest):
     tm.assert_categorical_equal(result, expected)
 
 
+def test_array_setitem_nullable_boolean_mask():
```

Comment:
It would be good to have a test in the base extensions tests as well (`base/setitem.py`), those can mimic the recently added tests in getitem.py (testing for nullable integer, nullable bool, the different errors, etc)

Reply:
i added a couple tests, not fully sure if it is what you mean, feel free to review, thanks @jorisvandenbossche